js/gameapp.js

```diff
@@ -175,21 +209,39 @@ gameSearchApp.controller('gameSearchCtrl', ['$http','$scope', function ($http, $
       return;
     }
     
+    if(search.length == 0){
+      invalidGameName.innerHTML = "You did not enter a game to search for!";
+      formButtons.innerHTML = '<a class="btn btn-primary" href="/">Try Again</a>';
+      return;
+    }
     
-    $scope.target= parseInt(document.getElementById('target').value,10);
-    
+    var target = document.getElementById('target').value;
+    /*
+    Adapted from a post on W3Resource: 
+    http://www.w3resource.com/javascript/form/all-numbers.php
+    */
+    var numbers = /^[0-9]+$/;
     var invalidNumber = document.getElementById('invalidNumber');
-    if (($scope.target < 0 ||$scope.target >= 101  ) || isNaN($scope.target )){
-      
-      invalidNumber.innerHTML = "Your threshold must be between 1 and 100";
+    
+    if(!target.match(numbers) && target.length != 0){
+      invalidNumber.innerHTML = "You did not type in a number.";
```

Comment:
changing html in your javascript is bad practice you should set a scope variable in your javascript and then use a ng-if or something similar in your .html file